src/net.cpp

```diff
@@ -1650,14 +1664,12 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
                 // We will not make outgoing connections to peers that are unreachable
                 // (e.g. because of -onlynet configuration).
                 // Therefore, we do not add them to addrman in the first place.
-                // Note that if you change -onlynet setting from one network to another,
-                // peers.dat will contain only peers of unreachable networks and
-                // manual intervention will be needed (either delete peers.dat after
-                // configuration change or manually add some reachable peer using addnode),
-                // see <https://github.com/bitcoin/bitcoin/issues/26035> for details.
+                // In case previously unreachable networks become reachable
+                // (e.g. in case of -onlynet changes by the user), fixed seeds will
+                // be loaded only for networks for which no we have no addressses.
                 seed_addrs.erase(std::remove_if(seed_addrs.begin(), seed_addrs.end(),
-                                               [](const CAddress& addr) { return !IsReachable(addr); }),
-                                seed_addrs.end());
+                                                [fixed_seed_networks](const CAddress& addr) { return std::find(fixed_seed_networks.begin(), fixed_seed_networks.end(), addr.GetNetwork()) == fixed_seed_networks.end(); }),
```

Comment:
This will iterate `fixed_seed_networks` for every seed address: `O(num_addresses * num_networks)`. I know the list of networks is tiny and this can probably be ignored wrt performance, but it can easily be made `O(num_addresses)`. Also, if `std::unordered_set` is used, the code will be shorter:

```cpp
std::find(fixed_seed_networks.begin(), fixed_seed_networks.end(), addr.GetNetwork()) == fixed_seed_networks.end();
```

would become:

```cpp
fixed_seed_networks.count(addr.GetNetwork()) == 0;
```

Reply:
changed it to unordered_set as suggested, thanks!